akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -744,6 +744,17 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
     shutdownCompletedPromise.future.map(_ => ())(system.dispatcher)
   }
 
+  /**
+   * Adds the connection pools to the actor system's coordinated shutdown, so that [[shutdownAllConnectionPools]] gets
+   * called before the actor system finally terminates.
+   */
+  def addClientPoolsToCoordinatedShutdown(): Unit = {
```

Comment:
I added this, but for client pools we might as well add it to coordinated shutdown automatically.